File: php/src/Model/CommentAuthor.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * `user` = console human, `worker` = on behalf of a worker, `api` = the integration.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class CommentAuthor
{
    public function __construct(
        public readonly string $type,
        public readonly ?string $id,
        public readonly ?string $label,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            type: (string) ($data['type'] ?? ''),
            id: isset($data['id']) ? (string) $data['id'] : null,
            label: isset($data['label']) ? (string) $data['label'] : null,
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/CommentAuthor.java
package io.fivexer.sdk.model;

/**
 * {@code user} = console human, {@code worker} = on behalf of a worker, {@code api} = the integration.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class CommentAuthor {
    private String type;
    private String id;
    private String label;

    public String getType() { return type; }
    public String getId() { return id; }
    public String getLabel() { return label; }
}
